simln-lib/src/sim_node.rs

```diff
@@ -639,7 +639,7 @@ pub struct SimGraph {
     channels: Arc<Mutex<HashMap<ShortChannelID, SimulatedChannel>>>,
 
     /// track all tasks spawned to process payments in the graph.
-    tasks: JoinSet<()>,
+    tasks: TaskTracker,
```

Comment:
If we pass this in, add a comment that `close` / `wait` of these tasks must be externally managed.